src/controllers/notifications.js

```diff
@@ -71,6 +73,16 @@ const UnacceptableExpiryError = require('../errors/unacceptable-expiry-error')
 exports.post = function * postNotification () {
   const notification = yield requestUtil.validateBody(this, 'Notification')
   log.debug('Got notification: ' + JSON.stringify(notification))
+
+  const verifySignature = this.config.get('notifications.must_verify')
+  if (verifySignature) {
+    const result = model.verifySignature(notification, this.config)
+    if (!result.valid) {
+      log.warn('Signature verification failed: ' + result.error)
+      throw new UnprocessableEntityError('Notification failed signature verification')
```

Comment:
> If the load balancer or upstream server is misconfigured, retrying won't help.

Sure retrying will help. The next time you retry, the misconfiguration may have been fixed.

> Seeing many failed payments, though, might be an indicator that something went wrong.

With atomic mode, infinite retries and proper monitoring (each 4xx error between banks for instance should cause an immediate ops alert), there should never **be** a single prepared payment that fails.

> You could also add metadata to each failed notification indicating if it was retried and only retry on startup (i.e.g, ledger is running, notifications fail, in the db with retry=false. Ledger restarts, loads failed notifications with retry=false, re-sends them, sets retry=true).

That only helps if the misconfiguration is on the ledger side, not if its on the connector side.